File: SoftUni Lessons/Python Development/Python Basics April 2019/Past Exams and Problems/8 - 6 and 7 April 2019/05. Movie Ratings.py
import sys

count_movies = int(input())

max_rated_movie = -sys.maxsize - 1
min_rated_movie = sys.maxsize
max_rated_name = None
min_rated_name = None
total_rating = 0

for movie in range(count_movies):
    name = input()
    rating = float(input())
    total_rating += rating
    if rating > max_rated_movie:
        max_rated_movie = rating
        max_rated_name = name
    if rating < min_rated_movie:
        min_rated_movie = rating
        min_rated_name = name

print(f"{max_rated_name} is with highest rating: {max_rated_movie:.1f}")
print(f"{min_rated_name} is with lowest rating: {min_rated_movie:.1f}")
print(f"Average rating: {(total_rating / count_movies):.1f}")

File: SoftUni Lessons/JavaScript Development/JavaScript Basics March 2020/21. МИНАЛИ ИЗПИТИ/05. Online Exam - 6 and 7 April 2019/05. Movie Ratings.js
function movieRatings(input) {
    // Mask
    let count = Number(input.shift());
    
    let bestRatedName = '';
    let bestRating = Number.MIN_SAFE_INTEGER;
    let worstRatedName = '';
    let worstRating = Number.MAX_SAFE_INTEGER;
    let totalScores = 0;

    for(let movie = 1; movie <= count; movie += 1) {
        let name = input.shift();
        let rating = Number(input.shift());

        if (rating > bestRating) {
            bestRatedName = name;
            bestRating = rating;
        }

        if (rating < worstRating) {
            worstRatedName = name;
            worstRating = rating;
        }
    
        totalScores += rating;
    }

    console.log(`${bestRatedName} is with highest rating: ${bestRating.toFixed(1)}`);
    console.log(`${worstRatedName} is with lowest rating: ${worstRating.toFixed(1)}`);
    console.log(`Average rating: ${(totalScores / count).toFixed(1)}`);
}

movieRatings([5, 'A Star is Born', 7.8, 'Creed 2', 7.3, 'Mary Poppins', 7.2, 'Vice', 7.2, 'Captain Marvel', 7.1]);  
// Should return:
// A Star is Born is with highest rating: 7.8
// Captain Marvel is with lowest rating: 7.1
// Average rating: 7.3

movieRatings([3, 'Interstellar', 8.5, 'Dangal', 8.3, 'Green Book', 8.2]);  
// Should return:
// Interstellar is with highest rating: 8.5
// Green Book is with lowest rating: 8.2
// Average rating: 8.3
